.github/workflows/docker-build-publish.yml

```diff
@@ -0,0 +1,100 @@
+name: Docker Build & Publish
+
+on:
+  push:
+    branches:
+      - "**"
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]+-alpha.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]+-beta.[0-9]+"
+      - "v[0-9]+.[0-9]+.[0-9]+-rc[0-9]+"
+  pull_request:
```

Comment:
[Question]
Won't building and publishing docker images for every branch be too much?

Reply:
I think so, though after discuss it with the team, it's okay to keep a track for all the images (probably, we should clean them after X time in the future) as we currently are doing: [packages](https://github.com/celestiaorg/celestia-app/pkgs/container/celestia-app/versions?filters%5Bversion_type%5D=tagged).

Reply:
I see, thanks